### ISSUE TITLE
parsing error on centos platform

### ISSUE THREAD

**Issue by rootmout** [jqlang/jq#2488]
**Describe the bug**

In a centos environment, the `jq` binary installed from dnf does not work in the same way as the one downloaded directly from github [1.6 releases page](https://github.com/stedolan/jq/releases/tag/jq-1.6)

**To Reproduce**

:information_source: the dockerfile used is at the end of this issue.
```bash
user@laptop:~/$ podman run rootmout/debub-jq-parsing-error-in-centos:v1.0.0 cat entry_file.txt 
I am not a jsonee line
{"code":200,"message":"I am a json line","stack":"","level":"info"}
user@laptop:~/$ podman run rootmout/debub-jq-parsing-error-in-centos:v1.0.0 jq --version
jq-1.6
user@laptop:~/$ podman run rootmout/debub-jq-parsing-error-in-centos:v1.0.0 jq -R 'fromjson' entry_file.txt
jq: error (at entry_file.txt:1): Invalid numeric literal at line 1, column 2 (while parsing 'I am not a jsonee line')
jq: error (at entry_file.txt:2): Invalid numeric literal at line 1, column 12 (while parsing '{"code":200,"message":"I am a json line","stack":"","level":"info"}')
user@laptop:~/$ podman run rootmout/debub-jq-parsing-error-in-centos:v1.0.0 ./jq-linux64 --version
jq-1.6
user@laptop:~/$ podman run rootmout/debub-jq-parsing-error-in-centos:v1.0.0 ./jq-linux64 -R 'fromjson' entry_file.txt
jq: error (at entry_file.txt:1): Invalid numeric literal at line 1, column 2 (while parsing 'I am not a jsonee line')
{
  "code": 200,
  "message": "I am a json line",
  "stack": "",
  "level": "info"
}
```

`rootmout/debub-jq-parsing-error-in-centos:v1.0.0` contain the binary `jq` installed by dnf and the `./jq-linux64` installed with wget from github, both are in version 1.6 and use the same file as entry but one installed by dnf fail with a parsing error while other is working fine.

**Expected behavior**

Both binary should work the same way.

**Environment (please complete the following information):**
 - OS and Version: CentOS Stream release 8
 - jq version: 1.6

**Additional doc**

Dockerfile used to create `debub-jq-parsing-error-in-centos:v1.0.0` image used in "To Reproduce" section:
```dockerfile
FROM quay.io/centos/centos:stream8

RUN dnf -y update -d 10 && \
    dnf -y --nodocs install wget vim jq && \
    dnf clean all

RUN wget https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64

RUN chmod +x jq-linux64

RUN echo -e \
  'I am not a json line\n{"code":200,"message":"I am a json line","stack":"","level":"info"}' \
  > entry_file.txt
```

**Comment by wader** [comment]
Hi, i suspect it might have to do with https://github.com/stedolan/jq/pull/2400 and it seems like the author of that PR also has fixed it in redhats packaging of jq, see https://bugzilla.redhat.com/show_bug.cgi?id=2008983 and https://gitlab.com/redhat/centos-stream/rpms/jq/-/tree/c9s. But then wouldn't it be the other way around? dnf installed would work but 1.6 from release would fail? strange

**Comment by itchyny** [comment]
This bug was fixed by #2400 (c4d39c4d22f2b12225ca1b311708f7e084ad9ff8) on May 27, 2022.
